Sprint 1 Roundup - Initial prototype

MertonL opened the pull request:
Implemented Pages: 
- Homepage
- Topics List
- Topic Page 
- Asset Page
- Learner Journeys List
- Learner Journey Page

Not implemented features/pages:
- Burger menu
- Search
- Related Assets
- Alternate Asset formats
- Learner journey duration

asimonw commented:
@MertonL In the `TopicList` component, you're using a method `topicUrl` to compute the url based on the topic name. That absolutely works, but for this sort of situation, using a computed property is better, as it will only recompute when the input (here the topic name) actually changes, while a method will run every time a component gets rendered. You won't notice a difference here, but it's a good habit to get into.

And then even smaller: on the `topics/index` page, the `topic-list` tag in the template can be self-closing. I think that's a nice style to get into (you and @georgemillard agree), that empty tags should be self-closing. Also, since the `loadItem` method isn't used, maybe comment it out (or delete) for clarity?

More feedback might come.

asimonw commented:
The code in `toggleFullscreen` manipulates the DOM directly. I'd use Vue's tools to avoid that!

Some coding style ones: 
- break the attributes into several lines in the `FullscreenButton` component
- consistently use `v-bind:` or `:` for binding props and vars. Standard practice is `:`. Same for `v-on`, I'd go for `@` consistently. It removes some of the char-noise and follows standard practices.

MertonL commented:
@asimonw Have updated the PR with new commits, I have refactored the fullscreen button & setup prettier with eslint. The settings don't seem to play nicely with multiline attributes so I haven't enabled that:
```    
'vue/max-attributes-per-line': [2,
      {
        'singleline': 1,
        'multiline': {
          'max': 1,
          'allowFirstLine': false
        }
      }
   ],
```

It's a start, I've added `npm run lint-autofix` so it can fix errors automatically in the codebase. I've added a `pre-commit` hook locally and it seems to work nicely.
